symengine/llvm_double.cpp

```diff
@@ -898,37 +984,57 @@ void LLVMDoubleVisitor::loads(const std::string &s)
     func = (intptr_t)executionengine->getPointerToFunction(F);
 }
 
-void LLVMDoubleVisitor::bvisit(const Floor &x)
+void LLVMVisitor::bvisit(const Floor &x)
 {
     std::vector<llvm::Value *> args;
     llvm::Function *fun;
     args.push_back(apply(*x.get_arg()));
-    fun = get_double_intrinsic(llvm::Intrinsic::floor, 1, mod);
+    fun = get_float_intrinsic(get_float_type(&mod->getContext()),
+                              llvm::Intrinsic::floor, 1, mod);
     auto r = builder->CreateCall(fun, args);
     r->setTailCall(true);
     result_ = r;
 }
 
-void LLVMDoubleVisitor::bvisit(const Ceiling &x)
+void LLVMVisitor::bvisit(const Ceiling &x)
 {
     std::vector<llvm::Value *> args;
     llvm::Function *fun;
     args.push_back(apply(*x.get_arg()));
-    fun = get_double_intrinsic(llvm::Intrinsic::ceil, 1, mod);
+    fun = get_float_intrinsic(get_float_type(&mod->getContext()),
+                              llvm::Intrinsic::ceil, 1, mod);
     auto r = builder->CreateCall(fun, args);
     r->setTailCall(true);
     result_ = r;
 }
 
-void LLVMDoubleVisitor::bvisit(const Truncate &x)
+void LLVMVisitor::bvisit(const Truncate &x)
 {
     std::vector<llvm::Value *> args;
     llvm::Function *fun;
     args.push_back(apply(*x.get_arg()));
-    fun = get_double_intrinsic(llvm::Intrinsic::trunc, 1, mod);
+    fun = get_float_intrinsic(get_float_type(&mod->getContext()),
+                              llvm::Intrinsic::trunc, 1, mod);
     auto r = builder->CreateCall(fun, args);
     r->setTailCall(true);
     result_ = r;
 }
 
+llvm::Type *LLVMDoubleVisitor::get_float_type(llvm::LLVMContext *context)
+{
+    return llvm::Type::getDoubleTy(*context);
+}
+
+llvm::Type *LLVMFloatVisitor::get_float_type(llvm::LLVMContext *context)
+{
+    return llvm::Type::getFloatTy(*context);
+}
+
+#if defined(SYMENGINE_HAVE_LLVM_LONG_DOUBLE)
+llvm::Type *LLVMLongDoubleVisitor::get_float_type(llvm::LLVMContext *context)
+{
+    return llvm::Type::getX86_FP80Ty(*context);
```

Comment:
We don't have access to Power9 hardware to test their FP128 type right?

Reply:
So we're only supporting long double on x86 for now? Just checking, sounds painful to fix CI for something that's unofficially supported anyhow.